src/Runner/Game.java

```diff
@@ -3,99 +3,71 @@
-			enemy.setWeaponName(enemyWeapon);
-			
-			return enemy;
+	public BasePlayer createPlayer() {
+		final String profession = JOptionPane.showInputDialog(null,
+					"Pick a class!", "Class Selection",
+					JOptionPane.QUESTION_MESSAGE, null,
+					Professions.ALLOWED_PROFESSION_NAMES.toArray(), "select a class").toString();
+		if (!Professions.isValidProfession(profession)) {
+			throw new IllegalStateException("Invalid profession entered. This should never happen.");
 		}
 		
+		final String playerName = JOptionPane.showInputDialog(null, "Enter the name of your " + profession);
+		final String weaponName = JOptionPane.showInputDialog(null, "Enter the name of your Weapon");
 		
-		
-		
-		String classSelection = (String) JOptionPane.showInputDialog(null,"Pick a class!", "Class Selection",JOptionPane.QUESTION_MESSAGE,null,names,"select a class").toString();
-		String name = (String) JOptionPane.showInputDialog(null,"Enter the name of your " + classSelection);
-		String nameOfWeapon = (String) JOptionPane.showInputDialog(null,"Enter the name of your Weapon");
-		
-		BiFunction<String,String,Swordsman> createSwordsman = Swordsman::new;
-		BiFunction<String,String,Marksman> createMarksman = Marksman::new;
-		BiFunction<String,String,Tank> createTank = Tank::new;
-		BiFunction<String,String,Mage> createMage = Mage::new;
-			
-		createsCharacter.put(Names.SWORDSMAN.toString(), (BasePlayer) createSwordsman.apply(name,nameOfWeapon));
-		createsCharacter.put(Names.MARKSMAN.toString(), (BasePlayer) createMarksman.apply(name, nameOfWeapon));
-		createsCharacter.put(Names.TANK.toString(), (BasePlayer) createTank.apply(name, nameOfWeapon));
-		createsCharacter.put(Names.MAGE.toString(), (BasePlayer) createMage.apply(name, nameOfWeapon));
-			
-		BasePlayer character = createsCharacter.get(classSelection);
-
-		return character;
-		
+		return Professions.getPlayerForProfession(profession, playerName, weaponName);
 	}
 	
-
-	public static void fight(BasePlayer player, BasePlayer Enemy) {
-
-		System.out.print(player.getName() + " vs " + Enemy.getName());
+	public BasePlayer createEnemy() {
+		return Professions.getPlayerForProfession(Professions.getRandomProfession(this.rng), DEFAULT_ENEMY_NAME, DEFAULT_ENEMY_WEAPON_NAME);
+	}
+	
+	public void fight(final BasePlayer player, final BasePlayer enemy) {
+		System.out.print(player.getName() + " vs " + enemy.getName());
 		
-		int round = 1;
-		while(round > 0){
-			while(player.getHealth() > 0 || Enemy.getHealth() > 0){
-				Random Turn = new Random();
-				Random Turn2 = new Random();
-				int playerTurn = Turn.nextInt(100);
-				int enemyTurn = Turn2.nextInt(100);
-				
-				if(playerTurn > enemyTurn){
-					player.attack(Enemy);
-				}else{
-					Enemy.attack(player);
-				}
-			}
-			round = -2;
+		while (player.getHealth() > 0 && enemy.getHealth() > 0) {
+			final int roll = rng.nextInt(2); // 0 or 1
+			final BasePlayer attacker = roll == 0 ? player : enemy,
+							 defender = roll == 1 ? player : enemy;

```

Comment:
don't know what github did to my spacing here, this should've lined up underneath `attacker`